### DISCUSSION TITLE
My ugly one . . .

### DISCUSSION THREAD

**Discussion by Benouare** [kernc/backtesting.py#262] [Show and tell]
Hi, 

That's post is just to give you some ideas.

So, for my use, I decide to build an app based on django and the backtesting lib. 
With this, there is some custom wrapper to consume API to get last OHLC of the day for example. 

The index : 
![Capture d’écran 2021-02-22 à 23 36 03](https://user-images.githubusercontent.com/28023340/108780933-6c10fa80-7569-11eb-91f3-15764484c0b1.png)

So you can see a beautiful web 1.0 app, with beautiful colors (style="color:red"), and nice html table... 
About the menu : 
* "home"-> index
* All other stocks, is a web page about stocks that i am not currently analysing/trading. But, I am able to analyse those stocks because i have (or can download) the OHLC data, and I am able to execute my algos on it. 
* Current month : the stats of the current month (01/02 -> 31/02 for example),
* 1 month -> stats from 13/01 -> 13/02
* ....
* Generate all the time, redo for each stock, the backtest execution regarding the configuration of the stock. 
* Download -> download last quotes (normally, just the quote of the day, but if I add a new stock, the app will download all the OHLC availables) 

In the table, u can see different stocks that i am currently working with. There is some informations like if the trade is "open"(in blue)  or not (in white), if i must buy/sell a stock tomorrow (in orange), the current price of my TP/SL, the pl of the trade, the 6 month WinRate, 6 months return, 6 months  PV-MV, ... the name of the algo that is used for this stock, win rate from the begining, the win rate from the begining.... 

You can see some times that the PL/SL is in red, that's mean that at Day-1, it the SL/TP was X, and now I must change it to Y. It makes me able to see quickly and easily what stock i have to upgrade on my bank account. 

6 months stats : 
![Capture d’écran 2021-02-22 à 23 41 17](https://user-images.githubusercontent.com/28023340/108780950-6fa48180-7569-11eb-979e-98fec1955d72.png)
I think it doesnt needs comment. That's just a view that makes me able to see quickly some stats about all my stocks.

This view makes me able to see all other stocks that i am not working with. This view makes me able to know witch stock could be interesting. It makes me able to make kind of pre-selection of a stock that I will analyse deeper after.
![Capture d’écran 2021-02-22 à 23 50 02](https://user-images.githubusercontent.com/28023340/108780952-703d1800-7569-11eb-99fa-ba24af4d0c05.png)

Here, the details of the performance of the stocks with all my algos (last column, the strat name)
![Capture d’écran 2021-02-22 à 23 53 37](https://user-images.githubusercontent.com/28023340/108780955-70d5ae80-7569-11eb-8396-be50792edd35.png)

That's a quick and dirty app, but in my case, that's making the job.

**Comment by kernc** [comment]
Nice!

I wonder what the names natixis, fagron, wendel, solvay ... signify. Are these strategies based on news / business networks? :sweat_smile: 

How long does it all take to recompute?

> ![Untitled](https://user-images.githubusercontent.com/684364/109240005-ea260900-77d6-11eb-9947-7d89a593b146.png)

Based on 9.04e7 return over 13 years, how leveraged is this? 50% temporal exposure, with those returns, indicates the strategy is clearly overfit! :stuck_out_tongue:

**Reply by Benouare** [reply]
On the last print screen, the natixis_v, fagron_v, wendel_v,  are strategies. 
On others print screens, they are mainly stocks names or strategies (depending of the column)
Those names are like that because I ve started those strategies on the stock that they have the same name. Just because of that. Those names are really arbitrary, it could be called Rabbit, Blue, Green.
Some are like turtule soup strat with some specific parameters, some are based on other information. But they are currently not based on news/business networks at all. 

If i recompute only stocks (ex : Apple, Google, Exxon, Amazon, Fb) that i am trading (first screen), with only one strategie (Apple=>Strat_bbb_v1, Google=>Strat_aaa_v3,...) for each, it takes something like ... 2 min?
If, for a stock (ex  : Apple),  i am running all my strats (Strat_aaa_v3,Strat_bbb_v1,....) it will takes something like 5 min for each stocks.

> Based on 9.04e7 return over 13 years, how leveraged is this? 50% temporal exposure, with those returns, indicates the strategy is clearly overfit! 😛


You are right, this one looks more realistic : 
![Capture d’écran 2021-02-26 à 09 18 01](https://user-images.githubusercontent.com/28023340/109276500-625fef00-7816-11eb-8ae0-d2f085f094a1.png)
